addons/project_todo/models/project_task.py

```diff
@@ -52,3 +52,18 @@ def action_convert_to_task(self):
             'res_id': self.id,
             'type': 'ir.actions.act_window',
         }
+
+    @api.model
+    def get_todo_views_id(self):
+        """ Returns the ids of the main views used in the To-Do app.
+
+        :return: a list of views id and views type
+                 e.g. [(kanban_view_id, "kanban"), (list_view_id, "list"), ...]
+        :rtype: list(tuple())
+        """
+        return [
+            (self.env.ref("project_todo.project_task_view_todo_kanban").id, "kanban"),
+            (self.env.ref("project_todo.project_task_view_todo_tree").id, "list"),
+            (self.env.ref("project_todo.project_task_view_todo_form").id, "form"),
+            (self.env.ref("project_todo.project_task_view_todo_activity").id, "activity"),
+        ]
```

Comment:
We could use `_xmlid_to_res_id` method from `ir.model.data` model to avoid browsing the recordset of each view to just have the id. 🙂 

```suggestion
        return [
            (self.env['ir.model.data']._xmlid_to_res_id("project_todo.project_task_view_todo_kanban").id, "kanban"),
            (self.env['ir.model.data']._xmlid_to_res_id("project_todo.project_task_view_todo_tree").id, "list"),
            (self.env['ir.model.data']._xmlid_to_res_id("project_todo.project_task_view_todo_form").id, "form"),
            (self.env['ir.model.data']._xmlid_to_res_id("project_todo.project_task_view_todo_activity").id, "activity"),
        ]
```

Reply:
Without the `.id` then I guess :)

Reply:
oups indeed :D